scene/gui/control.cpp

```diff
@@ -2307,6 +2307,11 @@ Transform2D Control::get_transform() const {
 	return xform;
 }
 
+void Control::set_transform(const Transform2D &p_transform) {
+	// control doesn't support setting transform manually yet
+	ERR_FAIL_MSG("Control does not support setting transform directly.  Use set_position(), set_scale(), etc.");
```

Comment:
It doesn't make sense for this method to exist if it will just always fail.

As for whether we can implement this... yeah, it can be done. But is that desired? Control nodes usually are not meant to be used this way. If you want full control over a canvas node's transform, Node2D would make more sense.

Reply:
It has to exist or the engine will not compile, as set_transform is now pure virtual in the base class.  It's not exposed anywhere, and can hopefully be used if CanvasItem is re-written anytime to use a transform instead of 3 other positional properties (outside the scope of this PR).

The alternative is to make set_transform be non pure-virtual and implement a noop in CanvasItem, which effectively does the same thing, except if you ever use set_transform on a Control you will be un-aware of the issue, which I why i implemented it this way with a warning.

Reply:
The alternative would be to define `set_transform` on Node2D and not on CanvasItem :)

Reply:
It's been like that for 7 years, since the first commit...  That's why the suggestion to correct.  There is no good reason why these aren't in CanvasItem, and if they are in the base class anyone can use recursion to the traverse up the tree, etc.   If my class inherits from CanvasItem, having these in Node2D don't help much do they

Reply:
Currently the only two classes in-engine classes that inherit from CanvasItem are Control and Node2D, surely we can fix some of the quirks here.